src/xenia/kernel/xam/xam_user.cc

```diff
@@ -287,8 +308,56 @@ SHIM_CALL XamUserWriteProfileSettings_shim(PPCContext* ppc_context,
     return;
   }
 
-  // TODO(benvanik): update and save settings.
-  // const auto& user_profile = kernel_state->user_profile();
+  // Update and save settings.
+  const auto& user_profile = kernel_state->user_profile();
+  auto settings_data_arr = reinterpret_cast<X_USER_WRITE_PROFILE_SETTING*>(
+      SHIM_MEM_ADDR(settings_ptr));
+
+  for (uint32_t n = 0; n < setting_count; ++n) {
+    const X_USER_WRITE_PROFILE_SETTING& settings_data = settings_data_arr[n];
+    XELOGD(
+        "XamUserWriteProfileSettings: setting index [%d]:"
+        " from=%d setting_id=%.8X data.type=%d",
+        n, (uint32_t)settings_data.from, (uint32_t)settings_data.setting_id,
+        settings_data.type);
+
+    xam::UserProfile::Setting::Type settingType =
```

Comment:
tip: auto in c++11 is nifty, makes this:

```
auto settingType = static_cast<xam::UserProfile::Setting::Type>(settings_data.type);
```

(as the static_cast names the type already)